### PR TITLE
Execute migration statements in order

### LINKED ISSUE #84: Previously working migration now fails
After upgrading to crystal 0.24.1 and bumping all the lucky versions
```yaml
  lucky_migrator:
    github: luckyframework/lucky_migrator
    version: ~> 0.4.0
```
one migration of my blog app doesn't run anymore. It fails with `column "slug" of relation "posts" does not exist`.

The migration that fails is:
```crystal
class AddSlugToPosts::V20171220204327 < LuckyMigrator::Migration::V1
  def migrate
    alter :posts do
      add slug : String?
    end

    # the following is the offending line, if I remove it, it works
    execute "UPDATE posts SET slug = regexp_replace(lower(title), '\\s', '-', 'g')"
  end
end
```

Looks like something changed regarding the sequence of adding the column and executing the raw sql?

### PR DESCRIPTION
**Pull request by mikeeus**
Closes #84 

When I changed the `Migration` to execute statements in a transaction, I collected them in `@prepared_statements` then executed them in order. But the `Migrations#execute` method executed the statement outside of the transaction as soon as it was called. This PR fixes that.

The one catch is that I had to restrict the `execute` method to strings. I could have allows it to accept `*args`, but that requires building the statement by escaping the the arguments manually, or significantly changing the `@prepared_statements` array and `execute_in_transaction` to accommodate it.

Lastly, I wrote a spec to catch this error, but it's dependent on `LuckyRecord::table_columns` which isn't the `LuckyRecord` version used by `LuckyMigrator`. Should I update the dependencies in a separate PR?

### CODE REVIEW

File: src/lucky_migrator/migration.cr [hunk 1]
```diff
@@ -70,10 +70,8 @@ abstract class LuckyMigrator::Migration::V1
     tx.connection.exec "DELETE FROM migrations WHERE version = $1", version
   end
 
-  private def execute(*args)
-    DB.open(LuckyRecord::Repo.settings.url) do |db|
-      db.exec *args
-    end
+  private def execute(statement : String)
```
Comment (paulcsmith) [inline src/lucky_migrator/migration.cr]: I think restricting by String is fine 👍

File: spec/migration_spec.cr [hunk 2]
```diff
@@ -26,4 +41,17 @@ describe LuckyMigrator::Migration::V1 do
...
   end
+
+  pending "statement execution order" do
+    Spec.after_each do
+      LuckyRecord::Repo.db.exec "DROP TABLE execution_order;"
+    end
+
+    it "runs execute statements in the order they were called" do
+      MigrationWithOrderDependentExecute::V998.new.up
+      columns = LuckyRecord::Repo.table_columns("execution_order")
```
Comment (paulcsmith) [inline spec/migration_spec.cr]: Feel free to update the dependency in this PR

Comment (mikeeus) [reply spec/migration_spec.cr]: Sorry I was mistaken. The `LuckyRecord::Repo.table_columns` method isn't in the latest release of `LuckyRecord`

Comment (paulcsmith) [reply spec/migration_spec.cr]: Would it make sense to run some manual SQL to get the column names?

Comment (mikeeus) [reply spec/migration_spec.cr]: Yeah definitely. I added a method in the spec.

I also added arguments to `Migration#up` and `Migration#down` to disable logging so its quiet during the spec.

File: spec/migration_spec.cr [hunk 3]
```diff
@@ -26,4 +42,28 @@ describe LuckyMigrator::Migration::V1 do
...
+    it "runs execute statements in the order they were called" do
+      MigrationWithOrderDependentExecute::V998.new.up(log: false)
+      columns = get_column_names("execution_order")
+      columns.includes?("new_col").should be_true
+      columns.includes?("bar").should be_true
+    end
+  end
+end
+
+def get_column_names(table_name)
```
Comment (paulcsmith) [inline spec/migration_spec.cr]: Could you add `private` to this so it isn't exposed to other tests?

Comment (mikeeus) [reply spec/migration_spec.cr]: Sure

**Comment by paulcsmith** [comment]
Thanks for tackling this so quickly!

**Comment by mikeeus** [comment]
no problem! :)

**Pull request merged**